harness/determined/cli/experiment.py

```diff
@@ -111,8 +111,67 @@ def read_git_metadata(model_def_path: pathlib.Path) -> Tuple[str, str, str, str]
     return (remote_url, commit_hash, committer, commit_date)
 
 
+def _format_yaml_error(e: Exception, file: str) -> str:
+    """Formats a ruamel.yaml Exception so that the user does not see a traceback of our internal
+    APIs.
+
+
+    ---------------------------------------------------------------------------------------------
+    DuplicateKeyError Example:
+    Input:
+        Traceback (most recent call last):
+        ...
+        ruamel.yaml.constructor.DuplicateKeyError: while constructing a mapping
+        in "<unicode string>", line 1, column 1:
+            description: constrained_adaptiv ...
+            ^ (line: 1)
+        found duplicate key "checkpoint_storage" with value "{}" (original value: "{}")
+        in "<unicode string>", line 7, column 1:
+            checkpoint_storage:
+            ^ (line: 7)
+
+        To suppress this check see:
+            http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys
+
+        Duplicate keys will become an error in future releases, and are errors
+        by default when using the new API.
+
+        Failed to create experiment
+    Output:
+        Error: failed while loading experiment config file constrained_adaptive.yaml.
+        Duplicate Key Error: found duplicate key "learning_rate" with value "0.022"
+        (original value: "0.025")
+        in examples/hp_search_benchmarks/darts_cifar10_pytorch/constrained_adaptive.yaml,
+        line 23, column 3:
+            learning_rate: 0.022
+            ^ (line: 23)
+    ---------------------------------------------------------------------------------------------
+    """
+    if isinstance(e, yaml.constructor.DuplicateKeyError):
+        err_msg = (
+            "found"
+            + str(e)
+            .split("found")[1]  # Isolate part of error msg that says where the duplicate key is
```

Comment:
Whoa, sorry.  I did suggest printing a better message, but I don't think parsing another library's output is an appropriate strategy.  Honestly better to just leave it, or pre-detect the issue.